### ISSUE TITLE
"/" to chat does not work.

### ISSUE THREAD

**Issue by xeonxlip** [vinegarhq/sober#254]
### Acknowledgement of checking existent issues

- [X] I have checked if my issue has already been reported, and I am certain that my problem has not already been addressed.

### What problem did you encounter?

While using sober, the "/" key does not properly bring up the chat box. You can still chat by clicking the chat icon, then clicking the chat area, but it'd be nice to have this shortcut working again. Thanks in advance :)

**Comment by tristann21** [comment]
For me it appears to only affect certain games

Works fine on pressure, but doesn't on flee the facility

**Comment by dingusreal** [comment]
This happens to me too. I notice it happens only on games that still use Legacy Chat, and not games that use the new TextChatService.

**Comment by xeonxlip** [comment]
> This happens to me too. I notice it happens only on games that still use Legacy Chat, and not games that use the new TextChatService.

seems to be the case on my end as well, should've done testing in more games before bug reporting

**Comment by jkibort928** [comment]
Can confirm, happens to me on 3008 (https://www.roblox.com/games/2768379856/3008-2-73), which I assume uses legacy chat? Don't know how to confirm this.

**Comment by Cube9214** [comment]
> This happens to me too. I notice it happens only on games that still use Legacy Chat, and not games that use the new TextChatService.
seems to be correct, i did a bit of testing, and legacy chat is the problem.

**Comment by librelol** [comment]
I don't think this is a Sober issue rather it is a Roblox issue.

**Comment by totaltommygun** [comment]
> I don't think this is a Sober issue rather it is a Roblox issue.

The slash key is broken on Roblox on Windows as well??

**Comment by jkibort928** [comment]
I have just tested it on my android phone with a keyboard connected. Same issue. It looks like a mobile-specific issue, because it works fine on PC. Super unfortunate for Sober, I guess we have to wait for the Roblox gods to hear our pleas :(

**Comment by librelol** [comment]
> > I don't think this is a Sober issue rather it is a Roblox issue.
> 
> The slash key is broken on Roblox for Windows as well??

No, on old games in roblox which do not use the new chat module dont support commands. As the new chat module was the one to introduce the built in /commands autocomplete.

**Comment by librelol** [comment]
It doesn't relate to a specific platform either I believe. I haven't tested this recently but just from my experience I had to port over some old games to use the new chat commands.

**Comment by yoshiweegee** [comment]
> It looks like a mobile-specific issue, because it works fine on a Windows PC

Doesn't this mean this issue should be closed?

**Comment by librelol** [comment]
> > It looks like a mobile-specific issue, because it works fine on a Windows PC
> 
> Doesn't this mean this issue should be closed?

Yeah, probably should be closed.

**Comment by Wingysam** [comment]
I've updated my game to focus the legacy chat bar input when you press /. It's pretty simple, you just use ContextActionService to bind Enum.KeyCode.Slash to a function that calls TextBox#CaptureFocus on the relevant TextBox.

**Comment by blaze-developer** [comment]
> I've updated my game to focus the legacy chat bar input when you press /. It's pretty simple, you just use ContextActionService to bind Enum.KeyCode.Slash to a function that calls TextBox#CaptureFocus on the relevant TextBox.

How have you managed to do that?

**Comment by skoove** [comment]
I have made a bug report to roblox, including a link to @Wingysam's [reply](https://github.com/vinegarhq/sober/issues/254#issuecomment-2395431875)

**Comment by Wingysam** [comment]
@blaze-developer I can't share the code because it's bound by an NDA but it should be possible to build something similar from my description.

**Comment by blaze-developer** [comment]
> @blaze-developer I can't share the code because it's bound by an NDA but it should be possible to build something similar from my description. 

How is a client modification you did yourself bound by an NDA?

**Comment by Wingysam** [comment]
The codebase of [my game](https:/galaxy.casa) is NDA-bound.

**Comment by jkibort928** [comment]
> > I've updated my game to focus the legacy chat bar input when you press /. It's pretty simple, you just use ContextActionService to bind Enum.KeyCode.Slash to a function that calls TextBox#CaptureFocus on the relevant TextBox.
> 
> How have you managed to do that?

From my knowledge of Roblox and some quick googling, I believe you can just use UserInputService (for key detection) and have it use the GetTextBox method of the client-side chat modules documented here:(https://create.roblox.com/docs/chat/legacy/client-side-chat-modules)

Then it's a matter of focusing the textbox (https://developer.roblox.com/en-us/api-reference/function/TextBox/CaptureFocus)

**Comment by blaze-developer** [comment]
Perhaps I misunderstood you. I had assumed that you had modified your *client* to add a script that allowed this to work. I wasnt aware this was to do with a specific *game* on Roblox because this thread is about the Roblox *client* and should apply to all games as players, not as developers.

**Comment by Wingysam** [comment]
Yes, I meant that I added a client-side script to the game I develop. It makes Galaxy's chat usable on Sober, but does not help with other games.

**Comment by skoove** [comment]
Roblox has closed my bug report with this:

> Please know that Roblox does not encourage the use of peripheral devices. If you are using any, it may not work properly with the Roblox site, Client, or Studio.
> 
> If you require assistance with anything else, please write back to us.

**Comment by Wingysam** [comment]
Yeah I guess it's up to the individual game developers to work around the bug. It's simple to make it work but does require the dev to care.

**Comment by totaltommygun** [comment]
> Roblox has closed my bug report with this:
> 
> > Please know that Roblox does not encourage the use of peripheral devices. If you are using any, it may not work properly with the Roblox site, Client, or Studio.
> > If you require assistance with anything else, please write back to us.

Roblox won't care they're actually so lazy when it comes to this stuff. I submitted a bug report because the "more info" menu for Roblox on PS4, (the one that has the buttons to access private servers and game passes) was just one huge button and I couldn't select any of the sub-menu's. Despite sending them a video it was a several email back and fourth trying to "prove" to them it's not my controller nor anything on my end before they finally fixed it

**Comment by blaze-developer** [comment]
In the meantime, why can't sober kind of hook into the game and add an event listener?

**Comment by claymorwan** [comment]
Rooms and Doors doesn't work either, means I cant type in the chat since it's a first person game

**Comment by BlueStaggo** [comment]
Still broken. Example experience where pressing "/" does not work: [Classic Sonic Simulator Test Server](https://www.roblox.com/games/2925613126/CSS-TEST-SERVER)

**Comment by jkibort928** [comment]
> Still broken. Example experience where pressing "/" does not work: [Classic Sonic Simulator Test Server](https://www.roblox.com/games/2925613126/CSS-TEST-SERVER)

The best we can hope to do is report it to Roblox. It's an issue with the mobile version of the game. I launched it on my phone and plugged in a keyboard and noticed the same thing. Slash just doesn't open chat in those games.

**Comment by blaze-developer** [comment]
In the meantime, why cant the sober client listen for that keypress event and open the chat on the client? Its a client-side error so it would only make sense that while you wait for an official fix, a *custom* client could be used  to *fix* the error?

**Comment by tunis4** [comment]
> In the meantime, why cant the sober client listen for that keypress event and open the chat on the client? Its a client-side error so it would only make sense that while you wait for an official fix, a _custom_ client could be used to _fix_ the error?

if it was that easy we would have done it already. i spent days looking for a way to externally open the chat. they have a weird half broken special input system solely for the purpose of opening the old chat and there is no code compiled for firing that input event because they likely forgot to add it. it would take way too much effort to figure out and setup whats necessary to fire the event manually. also going that deep is risky because it could trigger exploit detections. so its not happening.

**Comment by BoogieHasFun** [comment]
![Screenshot from 2025-01-11 14-48-48](https://github.com/user-attachments/assets/4a55322f-a812-4d33-af42-be33f35c506a)

i cannot use the chat in [RBWR](https://www.roblox.com/games/11765852158/V1-5-0-Realistic-Boiling-Water-Reactor-Simulator). this game has no chatbox (as in you cannot see chat history) and the chats are local based on what room you're in, unit 1 cannot see chats from unit 2 unless they call.

**Comment by blaze-developer** [comment]
> > In the meantime, why cant the sober client listen for that keypress event and open the chat on the client? Its a client-side error so it would only make sense that while you wait for an official fix, a _custom_ client could be used to _fix_ the error?
> 
> if it was that easy we would have done it already. i spent days looking for a way to externally open the chat. they have a weird half broken special input system solely for the purpose of opening the old chat and there is no code compiled for firing that input event because they likely forgot to add it. it would take way too much effort to figure out and setup whats necessary to fire the event manually. also going that deep is risky because it could trigger exploit detections. so its not happening.

Maybe you could make the client actually click on the button when you press that keybind? Just an idea, thanks :3

**Comment by V3L0C1T13S** [comment]
> Maybe you could make the client actually click on the button when you press that keybind? Just an idea, thanks :3

Still not a great workaround. Not only can we not just "detect" legacy chat to avoid conflicting with custom chat systems and the new TextChatService based one, but legacy chat positions and sizes aren't guaranteed, and we can't detect when the chat is open, nor can we get the position or size of the chat. The best thing you can do is wait until Roblox removes the legacy chat system entirely, which, [according to Roblox](https://devforum.roblox.com/t/migrate-to-textchatservice-removing-support-for-legacy-chat-and-custom-chat-systems/3237100), should be by April 30th, 2025.

**Comment by blaze-developer** [comment]
Oh ok :)

**Comment by notwithering** [comment]
It's been over a month since Roblox has "removed" TextChatService yet it's still in many games. I'm unsure how Sober works, but I propose some sort of side loaded script. I've mocked up a quick script that works in studio which you should just be able to throw into sober, assuming it works like that.

```lua
local uis = game:GetService("UserInputService")
local plr = game.Players.LocalPlayer

if not plr.PlayerScripts:FindFirstChild("ChatScript") then
	-- not using legacy chat service
	return
end

local chat = require(plr.PlayerScripts.ChatScript.ChatMain)

uis.InputEnded:Connect(function(input: InputObject)
	if input.KeyCode ~= Enum.KeyCode.Slash then
		return
	end

	chat:FocusChatBar()
end)
```

https://github.com/user-attachments/assets/f5a1d47a-0964-4213-9328-b6fc9052d513

**Comment by blaze-developer** [comment]
> Sober is not a cheat, and so we can't just load Luau scripts into the client. If experience developers want to fix the issue, they should stop using legacy chat.

If something like this is not possible, why would the source code be sensitive?

If it can't load scripts, there is no value in hiding the source code, because there can be no malicious use of it.

Seemingly, either

A: sober cannot load scripts, therefore the source code is not hidden for the reason I've been told, or 

B: sober can load scripts, which would provide reason for closed source, but would contradict your statement.

Not to poke and pry, just pointing out a seeming logical inconsistency, or a misunderstanding. Could somebody clear this up for me? Because logically right now from my understanding, only one of the two statements could be true.

**Comment by notwithering** [comment]
> If it can't load scripts, there is no value in hiding the source code, because there can be no malicious use of it.

I've also been told that Sober is closed source because it can load scripts which I guess was a rumor. But I would assume it's closed source as the code is not "production ready" or something like that.

**Comment by Wingysam** [comment]
I think it could easily be made to load scripts, but the maintainers of it choose not to do that because it would make Sober a cheat.

**Comment by V3L0C1T13S** [comment]
> > Sober is not a cheat, and so we can't just load Luau scripts into the client. If experience developers want to fix the issue, they should stop using legacy chat.
> 
> If something like this is not possible, why would the source code be sensitive?
> 
> If it can't load scripts, there is no value in hiding the source code, because there can be no malicious use of it.
> 
> Seemingly, either
> 
> A: sober cannot load scripts, therefore the source code is not hidden for the reason I've been told, or
> 
> B: sober can load scripts, which would provide reason for closed source, but would contradict your statement.
> 
> Not to poke and pry, just pointing out a seeming logical inconsistency, or a misunderstanding. Could somebody clear this up for me? Because logically right now from my understanding, only one of the two statements could be true.

We could do this very easily, actually, but we don't want to because it'll likely get us on Roblox's bad side (and it's not our job to fix their Lua code). There's lots of other low level details that are too boring to get into, but the gist of it is that if we release the source code, Roblox would have many reasons to block us, and potentially any other methods of running Roblox on Linux in the future.
